packages/devtools_app/lib/src/screens/performance/flutter_frames_chart.dart

```diff
@@ -361,48 +335,71 @@ class FlutterFramesChartItem extends StatelessWidget {
-              color: selected ? colorScheme.selectedFrameBackgroundColor : null,
-              child: Column(
-                children: [
-                  // Dummy child so that the InkWell does not take up the entire column.
-                  const Expanded(child: SizedBox()),
-                  Row(
-                    crossAxisAlignment: CrossAxisAlignment.end,
-                    children: [
-                      ui,
-                      raster,
-                    ],
-                  ),
-                ],
+    final content = Padding(
+      padding: EdgeInsets.only(
+        bottom: _FlutterFramesChartState._frameNumberSectionHeight,
+      ),
+      child: InkWell(
+        onTap: _selectFrame,
+        child: Stack(
+          children: [
+            // TODO(kenz): make tooltip to persist if the frame is selected.
+            FlutterFrameTooltip(
+              frame: frame,
+              hasShaderJank: hasShaderJank,
+              child: Container(
+                padding: const EdgeInsets.symmetric(horizontal: densePadding),
+                color:
+                    selected ? colorScheme.selectedFrameBackgroundColor : null,
+                child: Column(
+                  children: [
+                    // Dummy child so that the InkWell does not take up the entire column.
+                    const Expanded(child: SizedBox()),
+                    Row(
+                      crossAxisAlignment: CrossAxisAlignment.end,
+                      children: [
+                        ui,
+                        raster,
+                      ],
+                    ),
+                  ],
+                ),
               ),
             ),
-          ),
-          if (selected)
-            Container(
-              key: selectedFrameIndicatorKey,
-              color: defaultSelectionColor,
-              height: selectedIndicatorHeight,
-            ),
-          if (hasShaderJank)
-            const Padding(
-              padding: EdgeInsets.only(
-                top: FlutterFramesChartItem.selectedIndicatorHeight,
+            if (selected)
+              Container(
+                key: selectedFrameIndicatorKey,
+                color: defaultSelectionColor,
+                height: selectedIndicatorHeight,
               ),
-              child: ShaderJankWarningIcon(),
-            ),
-        ],
+            if (hasShaderJank)
+              const Padding(
+                padding: EdgeInsets.only(
+                  top: FlutterFramesChartItem.selectedIndicatorHeight,
+                ),
+                child: ShaderJankWarningIcon(),
+              ),
+          ],
+        ),
       ),
     );
+    return index % 2 == 0
+        ? Stack(
+            children: [
+              content,
+              Container(
+                padding: EdgeInsets.only(top: defaultChartHeight),
+                height: defaultChartHeight +
```

Comment:
giving this container `height: _FlutterFramesChartState._frameNumberSectionHeight` and using `margin:defaultChartHeight` instead of padding may make this more clear.

Or instead of using margin, you could wrap in a Positioned widget and place directly at the offset you want it in the stack

Reply:
I like it! done.